packages/x-tree-view/src/internals/plugins/useTreeViewItems/useTreeViewItems.types.ts

```diff
@@ -34,6 +34,12 @@ export interface UseTreeViewItemsPublicAPI<R extends {}> {
    * @returns {TreeViewBaseItem[]} The items in the tree.
    */
   getItemTree: () => TreeViewBaseItem[];
+  /**
+   * Toggle the disabled state of the item with the given id.
+   * @param {TreeViewItemId } itemId The id of the item to get the children of.
+   * @param {boolean } disabled true if the item should be disabled.
+   */
+  toggleDisabledItem: (itemId: TreeViewItemId, disabled: boolean) => void;
```

Comment:
Based on https://github.com/mui/mui-x/issues/14617, I suggest we always use an object anytime we have (or might have) other arguments that the `itemId`.

For `toggleDisabledItem`, I would follow the signature of `selectItem`:

```tsx
// If you have a better idea for the name of the method
apiRef.current.setIsItemDisabled({
  // The id of the item to disable or enable
  itemId,
  // If `true` the item will be disabled
  // If `false` the item will be enabled
  // If not defined, the item's new disable status will be the opposite of its current one
  shouldBeDisabled?/ boolean,
});
```

WDYT?